doc/GUIDE.md

```diff
@@ -85,6 +85,27 @@ function pointer types, and typedefs of struct or union types result in addition
 being generated. (All of these are discussed in more detail in the
 [Using The Code Generated By Jextract section](#using-the-code-generated-by-jextract)).
 
+Generally speaking, the code that jextract generates is dependent on the platform on which
+jextract is running. For example, when a C header file is processed by the C pre-processor,
+it is possible for code in the header file to detect the current platform using
+pre-processor directives, and expand to a different result based on that platform (see
+also the section on [pre-processor definitions](#preprocessor-definitions)). Additionally,
+different built in C types can have different formats depending on the platform (for
+example, the `long` type has different formats on Linux and Windows). Both of these are
+examples of things that can lead jextract to generate different outputs depending on the
+platform on which it is run. Care should be taken when generating bindings with jextract
+on one platform, and using those bindings on another platform, as the header files may
+contain platform-dependent code that can cause the generated code to misbehave when used
+on another platform.
+
+However, it is also possible for a C library to be written in such a way that it is not
+platform dependent: a so-called _portable_ library. Sharing the bindings generated for a
```

Comment:
Perhaps here we might say what moves portable libraries might do - such as use `long long` instead of `long`, or using sized numeric types e.g. `int32`.

Reply:
Added this: 

> These libraries, for instance, use data types that have the same format on all supported platforms (such as `long long` instead of `long`, or an explicitly-sized integer type such as `int64_t`).